luarules/gadgets/unit_seismic_dots_and_advanced_radar.lua

```diff
@@ -0,0 +1,363 @@
+-- The technical term here is Low-Discrepancy Sequence
+-- https://en.wikipedia.org/wiki/Low-discrepancy_sequence
+-- We choose the next location of the wobble based on the Golden Angle, the "most irrational" number
+-- To remove any clockwise rotation bias, on each wobble advance a random number of Golden Angle "steps", from a small set of possible "steps", the set selected to:
+-- A. Average to approximately 180 degree rotation per wobble
+-- B. Rotate more than 90 degrees per step
+-- https://github.com/beyond-all-reason/Beyond-All-Reason/pull/5693#discussion_r2296246178
+local goldenAngle = (2-(1 + math.sqrt(5))/2)*math.pi*2
+local goldenSteps = {1,4,7,9,12,14}
+-- cache sin and cos calcs for wobble rotation
+local mathSinCache = {}
+for ix = 1, #goldenSteps do
+    mathSinCache[ix] = mSin(goldenSteps[ix]*goldenAngle)
+end
+local mathCosCache = {}
+for ix = 1, #goldenSteps do
+    mathCosCache[ix] = mCos(goldenSteps[ix]*goldenAngle)
+end
+
+-- hyperparameter for wobble drift time for seismic dots, in slowUpdates (15 frame increments)
+-- 4 seems like reasonable minimum (2 second drift time), any faster looks weird to me.
+-- 15 is the drift time of standard radar wobble. Seismic is desired to wobble as fast or faster than radar, so this is a reasonable max drift time.  
+local minRandDriftTime = 4
+local maxRandDriftTime = 15
+
+-- How long a seismic dot should last after last seismic ping, in frames
+-- (after target leaves seismic range or stops moving)
+local seismicDotTime = 2 * Game.gameSpeed
+
+-- controls how often units search for advanced radars while in enemy radar
+local basicRadarUpdateRate = 2 * Game.gameSpeed
+
+-- controls drift time of radar dots in advanced radar, and when to re-check for advanced radars
+local advancedRadarUpdateRate = 7.5 * Game.gameSpeed
+
+-- individual unit value caches
+local unitSpeeds = {}
+local unitAllyTeamCache = {}
+local unitInEnemyRadar = {}
+local unitInSeismic = {}
+local unitPendingWobble = {}
+local advancedRadars = {}
+
+-- cache UnitDefs at start
+local unitDefSpeeds = {}
+local unitDefSeismicSignature = {}
+local unitDefRadarEmitHeight = {}
+local unitDefRadarRadius = {}
+local unitDefWobbleRatio = {}
+local unitDefIsHover = {}
+local unitDefNoWobble = {}
+for unitDefID, unitDef in pairs(UnitDefs) do
+
+	unitDefSpeeds[unitDefID] = UnitDefs[unitDefID].speed
+	unitDefSeismicSignature[unitDefID] = UnitDefs[unitDefID].seismicSignature
+
+	if UnitDefs[unitDefID].customParams then
+		if UnitDefs[unitDefID].customParams.advancedradarwobbleratio ~= nil then
+			unitDefRadarEmitHeight[unitDefID] = UnitDefs[unitDefID].radarEmitHeight
+			unitDefRadarRadius[unitDefID] = mMax(UnitDefs[unitDefID].radarRadius,UnitDefs[unitDefID].sonarRadius)
+			unitDefWobbleRatio[unitDefID] = UnitDefs[unitDefID].customParams.advancedradarwobbleratio
+		end
+
+		if UnitDefs[unitDefID].customParams.radarwobble == "none" then
+			unitDefNoWobble[unitDefID] = true
+		end
```

Comment:
this is now built into the engine https://github.com/beyond-all-reason/RecoilEngine/pull/1831